### ISSUE TITLE
Change in date serialization for JSON

### ISSUE THREAD

**Issue by liaden** [rails/rails#41042]
### Steps to reproduce
Upgrading from Ruby 2.7.1 to Ruby 3.0.0:

`render json: SomeTable.last.attributes`

Note: we are not specifying any json gems with this application.
```
 cat Gemfile.lock| grep json
      simplecov_json_formatter (~> 0.1)
    simplecov_json_formatter (0.1.2)
```

### Expected behavior
The json should include a value like `\"created_at\":\"2021-01-07T04:00:06.332Z\",\"updated_at\":\"2021-01-07T04:00:06.332Z\"`.

### Actual behavior
The json has: `\"created_at\":\"2021-01-07 14:36:35 UTC\",\"updated_at\":\"2021-01-07 14:36:35 UTC\"`.

### System configuration
**Rails version**:
6.0.3.4

**Ruby version**:
* Expected: 2.7.1
* Actual: 3.0.0

**Comment by intrip** [comment]
A [reproduction script](https://gist.github.com/rafaelfranca/dd0a2d32397fd9c3a596a349e6601309) would help.

**Comment by lfalcao** [comment]
I am not sure if this is not the expected behavior...  but one possible fix would be add in your application: `Time::DATE_FORMATS[:default] = '%Y-%m-%dT%H:%M:%S.%LZ'` to keep the previous date format.

Reproduction script: https://gist.github.com/lfalcao/bd95f0b6778163ca69bb154867fd3dfd cc/ @intrip

**Comment by zzak** [comment]
@liaden The proposed workaround seems reasonable, how do you feel?

I'm not sure I have enough context in this area, but I know @pixeltrix might be able to provide some valuable insight here. 🙏

**Comment by pixeltrix** [comment]
This is fixed by #39697 and is on 6-0-stable and all 6.1 releases - if you're intending to use Ruby 3.0 I'd strongly suggest upgrading to Rails 6.1 first.

**Comment by zzak** [comment]
Thanks for taking a look Andrew! ❤️
